Fix broken links

raballew opened the pull request:
The repository contained a large amount of dead or broken links. Dead links with no clear up-to-date reference have been removed, while broken links were fixed. CI check that looks for broken links has been added.

raballew commented:
@quentinlesceller It seems the rfc repository contains a large amount of dead links as well. I would suggest rethinking the way external resources are linked in both the rfc repository and this one as those links end up being published on the official website. Some links I fixed have been paywalled since they were added to the repository.

quentinlesceller commented on .dockerignore:
```diff
@@ -1,6 +0,0 @@
-.gitmodules
-.dockerignore
-.editorconfig
-Containerfile
```
Any reason why you chose to delete this file?

raballew replied:
Neither the GitHub CI Actions nor the way how the local container deployment works require a `.dockerignore` file.

```bash
docker build -t docs -f Containerfile .
```

There is only one file copied to the container image `COPY [ "requirements.txt", "./"]` and everything else is mounted as a RO volume in the subsequent `docker run ...` command. Therefore I removed the file or am I not seeing something obvious?

sesam replied:
When you run docker manually it copies everything, and .gitignore isn't helping. Careful use of Dockerfile and docker-compose.yml probably ensures that not the whole folder contents is copied + therefore used as basis to decide when rebuild is needed. I suggest keeping the dockerignore and having it list log files and unrelated files. I might've recommended this as well once when grin was very privacy focused, as any logfiles, rust binaries with local user folder names, and similar could risk leaking in to the docker image, and then leak out as uninitialized memory. I think something such happened once, just not - known to me - in combination to leak any PII.

raballew replied:
@sesam This `Containerfile` is only used for debugging purposes to test the website locally without pushing everthing to master just to see if things work but if the performance of the build process is too slow, because the context copied is too big or PII leaks are real, than I will include the `.dockerignore` file again.

quentinlesceller commented on docs/stylesheets/extra.css:
```diff
@@ -24,9 +23,9 @@ h1 {
   --md-code-bg-color: rgb(20, 20, 23); /* set code-blocks background color*/
   --md-footer-bg-color: rgb(24, 24, 24); /* set footer background color */
-  /* --md-typeset-a-color: #51c0f5; */
+  --md-typeset-a-color: rgb(233, 235, 252); /* set active and link font color */
```
Nicely done. There is still an issue with links which are into admonition parts. e.g.:
![Capture d’écran 2021-04-07 à 10 47 00](https://user-images.githubusercontent.com/11842328/113886359-9c65ce80-978e-11eb-95ec-d7f88c0401e6.png)
![Capture d’écran 2021-04-07 à 10 47 02](https://user-images.githubusercontent.com/11842328/113886361-9cfe6500-978e-11eb-8809-c3596377a313.png)
This can be done here or in a different PR.

raballew replied:
This PR is big enough, so that I will handle this in a different one.

quentinlesceller commented on docs/wiki/extra-documents/regarding-foundations.md:
```diff
@@ -13,7 +13,7 @@ September 13, 2018: There are no current or future plans to create an official G
 
 * **Bitcoin** Foundation has been [mired in controversies](https://en.wikipedia.org/wiki/Bitcoin_Foundation) regarding its leadership and received cease and desist letters from US institutions.
-* The **Tezos** Foundation ended up in a [vicious legal dispute](https://www.wired.com/story/tezos-blockchain-love-story-horror-story/) with the project founders whilst some $400m in funds at the time were locked.
+* The **Tezos** Foundation ended up in a vicious legal dispute with the project founders whilst some $400m in funds at the time were locked.
```
Why remove the link?

raballew replied:
At least from my end, this link is behind a paywall and therefore not accessible for me.

quentinlesceller replied:
Indeed.

quentinlesceller commented on docs/wiki/extra-documents/nomenclature-bs.md:
```diff
@@ -77,7 +77,7 @@ Sources: CCITT Working Group on Alphabets, October 6-9, 1964, CBI 67, box 3, fol
 and now some other resources I've found... and some symbols that stand out. There are lots of options and I am 100% of the belief that the perfect one can jump out at us like a mystical epiphany, and that bikeshedding isn't as arduous and arbitrary as it always seems to be.
 
-From this document: http://bora.uib.no/bitstream/handle/1956/10699/MUFI-Alphabetic-4-0.pdf?sequence=9&isAllowed=y
+From this document: https://folk.uib.no/hnooh/mufi/specs/MUFI-Alphabetic-4-0.pdf
```
This whole doc is a remain of the old wiki. Should probably be deleted. But fine for this PR.

quentinlesceller commented on docs/wiki/security-process/security-team.md:
```diff
@@ -8,11 +8,13 @@
 
 ## Summary
+
 [summary]: #summary
```
Are these lines even necessary? AFAIK linking to #summary works without this line.

quentinlesceller replied:
Ah it's the RFC...

raballew replied:
Yes, it is the RFC

quentinlesceller commented on docs/wiki/services/list-of-services.md:
```diff
@@ -20,7 +20,6 @@
 ## Block Explorers
 
 - [Grinscan](https://grinscan.net/)
-- [Blockscan](https://grin.blockscan.com/)
```
👍

quentinlesceller commented on docs/wiki/transactions/slates.md:
```diff
@@ -496,5 +499,5 @@ value of the `feat` field. Currently only present if `feat` is 2.
 * QR Code encoding of slates
-* [Armored slates]()
-* [Slatepack](https://github.com/j01tz/grin-rfcs/blob/slatepack/text/0000-slatepack.md)
+* Armored slates
```
This document should link to 0012-compact-slates.md

quentinlesceller reviewed (changes requested):
Quite a bit of work here. Thank you @raballew :). Small comments below but overall looks like a good cleanup + fix.

quentinlesceller commented:
> @quentinlesceller It seems the rfc repository contains a large amount of dead links as well. I would suggest rethinking the way external resources are linked in both the rfc repository and this one as those links end up being published on the official website. Some links I fixed have been paywalled since they were added to the repository.
>
100% agree with this. Unsure how to proceed though.


\
> Furthermore I would like to introduce a link checker that runs daily and verifies that all internal / external links still work. What do you think?

The link checker is awesome ! For now I think having a link checker on every PR is good enough. I'm not sure what kind of maintenance would be needed.

raballew commented:
@quentinlesceller I changed the link to the compact slates.

> > @quentinlesceller It seems the rfc repository contains a large amount of dead links as well. I would suggest rethinking the way external resources are linked in both the rfc repository and this one as those links end up being published on the official website. Some links I fixed have been paywalled since they were added to the repository.
> 
> 100% agree with this. Unsure how to proceed though.

I would cleanup the RFC repository first, by fixing or removing any dead links and maybe configure the linkchecker there as well for each PR and on a daily basis so that project owners get notified if some of the links break as websites go down e.g. This should already prevent most of the issues from happening. Also, a practice that worked for me quite well is using relative links only in the same directory and absolute paths for everything else. But this is just a personal preference.

Also, try using permalinks whenever possible and never link to a PR or a forked repository as they tend to be deleted quite often.

> > Furthermore I would like to introduce a link checker that runs daily and verifies that all internal / external links still work. What do you think?
> 
> The link checker is awesome ! For now I think having a link checker on every PR is good enough. I'm not sure what kind of maintenance would be needed.

Run it once a day to become notified if anything breaks at third party websites referenced.

Paouky commented:
Most of this PR is out of my scope, but I went over the dead links. There's a bunch of URLs that should lead to translations that are now dead because they were moved into this sub-directory https://github.com/mimblewimble/grin/tree/master/doc/translations

So, for example, the dead link `https://github.com/mimblewimble/grin/blob/master/doc/build_ES.md` should be changed to `https://github.com/mimblewimble/grin/blob/master/doc/translations/build_ES.md`

Can you make the modifications?
Thanks a lot for this!

raballew commented:
@Paouky Sure, I will try to add the falsely removed links. My question would be if this really adds value as you need someone that is proficient in that language to review everything and if something ever changes you will need to adjust it in multiple places.

sesam commented:
I'd also like to say my thanks for working in this! I noticed the PR touches lot's of files. For optimal developer satisfaction and speediness, it usually pays off to do PRs in smaller chunks. (I know, git and github isn't really very helpful towards this. Any ideas of good for tools, or interest in building one - pls hit me up!)

raballew commented:
@sesam Agreed! Not sure how to handle fixes of the same kind of issue in a lot of files. I decided to focus on one kind of issue (broken links, formatting, e.g.) and then raise a PR per kind of issue to keep everything related to this in the same place. I guess when there is a systemic issue PRs just tend to be bigger, but I can split them in smaller chunks whenever possible.

raballew commented:
@Paouky Fix broken translation links
@sesam Added .dockerignore file

Paouky commented:
@quentinlesceller What do you say? Ready for merge?

quentinlesceller commented on docs/getting-started/build.md:
```diff
@@ -1,6 +1,6 @@
 # Build
 
-> *Read this in other languages: [Español](https://github.com/mimblewimble/grin/blob/master/doc/build_ES.md), [Korean](https://github.com/mimblewimble/grin/blob/master/doc/build_KR.md), [日本語](https://github.com/mimblewimble/grin/blob/master/doc/build_JP.md), [简体中文](https://github.com/mimblewimble/grin/blob/master/doc/build_ZH-CN.md).*
+> *Read this in other languages: [Español](https://github.com/mimblewimble/grin/blob/master/doc/translations/build_ES.md), [Korean](https://github.com/mimblewimble/grin/blob/master/doc/translations/build_KR.md), [日本語](https://github.com/mimblewimble/grin/blob/master/doc//translations/build_JP.md), [简体中文](https://github.com/mimblewimble/grin/blob/master/doc//translations/build_ZH-CN.md).*
```
```suggestion
> *Read this in other languages: [Español](https://github.com/mimblewimble/grin/blob/master/doc/translations/build_ES.md), [Korean](https://github.com/mimblewimble/grin/blob/master/doc/translations/build_KR.md), [日本語](https://github.com/mimblewimble/grin/blob/master/doc/translations/build_JP.md), [简体中文](https://github.com/mimblewimble/grin/blob/master/doc/translations/build_ZH-CN.md).*
```

quentinlesceller reviewed (approved):
Small typo. Otherwise looking good.
